utilities/audit-search.py

```diff
@@ -0,0 +1,190 @@
+        db_args['password'] = self._cfg.get('nipapd', 'db_pass')
+        db_args['sslmode'] = self._cfg.get('nipapd', 'db_sslmode')
+        db_args['port'] = self._cfg.get('nipapd', 'db_port')
+        if db_args['host'] is not None and db_args['host'] == '':
+            db_args['host'] = None
+        for key in db_args.copy():
+            if db_args[key] is None:
+                del(db_args[key])
+
+        while True:
+            try:
+                self._con_pg = psycopg2.connect(**db_args)
+                self._curs_pg = self._con_pg.cursor(cursor_factory=psycopg2.extras.DictCursor)
+                self._register_inet()
+                psycopg2.extras.register_hstore(self._con_pg, globally=True, unicode=True)
+            except psycopg2.Error as exc:
+                raise "Error"
+            except psycopg2.Warning as warn:
+                raise "Warning: %s" % warn
+
+            break
+
+
+    def _register_inet(self, oid=None, conn_or_curs=None):
+        """ Create the INET type and an Inet adapter."""
+        from psycopg2 import extensions as _ext
+        if not oid:
+            oid = 869
+        _ext.INET = _ext.new_type((oid, ), "INET",
+                lambda data, cursor: data and Inet(data) or None)
+        _ext.register_type(_ext.INET, self._con_pg)
+        return _ext.INET
+
+
+    def _format_log_prefix(self, data):
+        if type(data) is not dict:
+            print("Not a dict")
+            raise
+        output = "{:<15} {}\n".format("Log id", data['id'])
+        output += "{:<15} {}\n".format("Author:", data['username'])
+        output += "{:<15} {}\n".format("Date:", data['timestamp'].strftime('%c'))
+        if re.search(".*attr:.*", data['description']):
+            res = re.match(r'(.*\d\b).*attr: (\{.*\})', data['description'])
+            description = res.group(1)
+            dataset = res.group(2)
+            output += "{:<15} {}\n".format("Description:", description)
+            parsed = eval(dataset)
+            output += "New data:\n"
+            for k,v in parsed.items():
+                output += "{:<16}{:<20}: {:<32}\n".format("",k, v)
+        else:
+            output += "{:<15} {}\n".format("Description:", data['description'])
+        return output
+
+
+    def search_log_prefix(self, prefix):
+        if self._get_afi(prefix) is None:
+            raise ValueError("Invalid Prefix")
+
+        sql = "SELECT * FROM ip_net_log"
+
+        sql += " WHERE prefix_prefix <<= %s"
+        data = (prefix,)
+
+        self._curs_pg.execute(sql, data)
+        res = list()
+        for row in self._curs_pg:
+            res.append(dict(row))
+
+        output = "Audit log for prefix {}\n".format(prefix)
+        for entry in res:
+            output += self._format_log_prefix(entry)
+            output += "\n"
+
+        return output
+
+
+
+a = AuditLog()
+
+if len(sys.argv) == 1:
+    print "Use with prefix to search in audit log"
```

Comment:
We try to use py3 syntax so that print would be `print()`. If you still want to use py2 then you can do `from __future__ import print_function` to just get the new print function.